docs/src/accessibility-testing-js.md

```diff
@@ -0,0 +1,292 @@
+
+You can find the rule IDs to pass to `disableRules()` in the `id` property of the violations you want to suppress. A [complete list of axe's rules](https://github.com/dequelabs/axe-core/blob/master/doc/rule-descriptions.md) can be found in `axe-core`'s documentation.
+
+```js
+test('should not have any accessibility violations outside of rules with known issues', async function({ page }) {
+  await page.goto('https://your-site.com/page-with-known-issues');
+
+  const accessibilityScanResults = await new AxeBuilder({ page })
+    .disableRules(['duplicate-id'])
+    .analyze();
+
+  expect(accessibilityScanResults.violations).toEqual([]);
+});
+```
+
+### Using snapshots to allow specific known issues
+
+If you would like to allow for a more granular set of known issues, you can use [Snapshots](./test-snapshots.md) to verify that a set of pre-existing violations has not changed. This approach avoids the downsides of using `AxeBuilder.exclude()` at the cost of slightly more complexity and fragility.
+
+Do not use a snapshot of the entire `accessibilityScanResults.violations` array. It contains implementation details of the elements in question, such as a snippet of their rendered HTML; if you include these in your snapshots, it will make your tests prone to breaking every time one of the components in question changes for an unrelated reason:
+
+```js
+// Don't do this! This is fragile.
+expect(accessibilityScanResults.violations).toMatchSnapshot();
+```
+
+Instead, create a *fingerprint* of the violation(s) in question that contains only enough information to uniquely identify the issue, and use a snapshot of the fingerprint:
+
+```js
+// This is less fragile than snapshotting the entire violations array.
+expect(violationFingerprints(accessibilityScanResults)).toMatchSnapshot();
+
+// my-test-utils.js
+function violationFingerprints(accessibilityScanResults) {
+    const violationFingerprints = accessibilityScanResults.violations.map(violation => ({
+        rule: violation.id,
+        // These are CSS selectors which uniquely identify each element with
+        // a violation of the rule in question.
+        targets: violation.nodes.map(node => node.target),
+    }));
+
+    return JSON.stringify(violationFingerprints, null, 2);
+}
+```
+
+## Exporting a report as a test attachment
+
+There are many tools and libraries which support generating standalone reports for axe scan results. You can use [`testInfo.attach()`](./api/class-testinfo#test-info-attach) to attach these reports to your test results, which [reporters](./test-reporters) can embed or link as part of your test output.
+
+The following example uses the [`axe-sarif-converter` package](http://npmjs.com/package/axe-sarif-converter) to produce a [SARIF report](https://sarifweb.azurewebsites.net/) and attach it to the test result:
```

Comment:
I think this needs a bit more details. This is the first time SARIF is mentioned in our docs. Let's say that it's an easy way to see axe results, and tell how to see the results from the produced SARIF json with some nice UI.

Reply:
This is a great point. I tried adding more context, but it felt like I was shoehorning in a concept that was only tangentially related to the guide as a whole. I also found that for the specific use case of a SARIF report, I don't actually think that using a test attachment like this is a great idea; most other tools that integrate with SARIF expect to read a standalone SARIF file, which attaching as a test result like this doesn't actually necessarily produce.

Instead, I rewrote the section in terms of attaching the axe scan results without going through a separate report library. I think that's a more realistic use case for a test attachment; that information can be useful for debugging test cases that aren't finding all the violations a user expects.